File: 0008-maximum-average-subarray-i/solution.go
package main

func findMaxAverage(nums []int, k int) float64 {
	windowSum := 0

	// First window
	for i := 0; i < k; i++ {
		windowSum += nums[i]
	}

	maxSum := windowSum

	// Slide the window
	for i := k; i < len(nums); i++ {
		windowSum += nums[i]
		windowSum -= nums[i-k]

		if windowSum > maxSum {
			maxSum = windowSum
		}
	}

	return float64(maxSum) / float64(k)
}

File: 0008-maximum-average-subarray-i/solution.swift
func findMaxAverage(_ nums: [Int], _ k: Int) -> Double {
    var windowSum = 0
    
    for i in 0..<k {
        windowSum += nums[i]
    }
    print("First window sum=", windowSum)
    
    var maxSum = windowSum

    for i in k..<nums.count {
        windowSum += nums[i] // add next
        windowSum -= nums[i - k] // subtract first
        maxSum = max(maxSum, windowSum)
    }
    return Double(maxSum) / Double(k)
}
